### PR TITLE
ArC - add debug logging for the request context manipulation

### PR DESCRIPTION
**Pull request by mkouba**
- activation, deactivation and destruction

### CODE REVIEW

**Comment by mkouba** [comment]
This could be really helpful when debugging issues like https://github.com/quarkusio/quarkus/issues/27229.

**Review by Ladicek — APPROVED** [review]
I'd personally say it should be trace logging, but other than that, looks fine.

**Comment by mkouba** [comment]
> I'd personally say it should be trace logging, but other than that, looks fine.

Ok, let's use the TRACE level. I don't see a big difference here but it makes sense in general.

**Comment by mkouba** [comment]
> I'd personally say it should be trace logging, but other than that, looks fine.

Done.

**Review by Ladicek — APPROVED** [review]

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -354,6 +354,18 @@ The request context is destroyed:
...
+
+You can set the `TRACE` level for the logger `io.quarkus.arc.requestContext` and try to analyze the log output afterwards.
+
+.`application.properties` Example
+[source,properties]
+----
+quarkus.log.category."io.quarkus.arc.requestContext".min-level=TRACE <1>
+quarkus.log.category."io.quarkus.arc.requestContext".level=TRACE
+----
+<1> You also need to adjust the minimum log level for the relevant catagory.
```
Comment (manovotn) [inline docs/src/main/asciidoc/cdi-reference.adoc:367]:
```suggestion
<1> You also need to adjust the minimum log level for the relevant category.
```

**Review by manovotn — APPROVED** [review]
Looks good, only added a typo correction.

File: docs/src/main/asciidoc/cdi-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/cdi-reference.adoc]: Done, thanks.

**Comment by mkouba** [comment]
@gastaldi do you happen to know if [those tests](https://github.com/quarkusio/quarkus/pull/27249#issuecomment-1216467188) are a known problem?

**Comment by gastaldi** [comment]
@mkouba never seen those fail before, maybe @michalszynkiewicz knows something?

**Comment by manovotn** [comment]
> Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project quarkus-cli: There was a timeout in the fork

Looks like some timeouts, unlikely to be flaky tests but I can see them across several PRs already. Did we cross some timeout limit set for CI?

**Comment by gastaldi** [comment]
Not sure. Now I am seeing these failures in another PRs too, looks like some commit broke it or certificates expired somewhere, I'll investigate

**Comment by michalszynkiewicz** [comment]
I think these may be the tests that use badssl.com that @geoand created a while back

**Comment by geoand** [comment]
Indeed.

There is a script that regenerates those certificates

**Pull request merged**